mcux/mcux-sdk/devices/MIMXRT798S/drivers/Kconfig

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+# driver_clock
+config MCUX_USE_driver_clock
+    bool "Clock Driver"
+    select MCUX_USE_driver_common
+    help
+      Clock Driver
+
+# driver_dsp
+config MCUX_USE_driver_dsp
+    bool "dsp Driver"
+    select MCUX_USE_driver_power
```

Comment:
Why do we include the `fsl_power` for the DSP?
The driver has a lot of ARM specific registers or functions.
When compiling for HiFi4 I get:
```

In file included from /opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:8:
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h: In function 'POWER_EnableDeepSleep':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h:735:5: error: 'SCB' undeclared (first use in this function)
  735 |     SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
      |     ^~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h:735:5: note: each undeclared identifier is reported only once for each function it appears in
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h:735:17: error: 'SCB_SCR_SLEEPDEEP_Msk' undeclared (first use in this function)
  735 |     SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
      |                 ^~~~~~~~~~~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h: In function 'POWER_DisableDeepSleep':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h:743:5: error: 'SCB' undeclared (first use in this function)
  743 |     SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
      |     ^~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.h:743:18: error: 'SCB_SCR_SLEEPDEEP_Msk' undeclared (first use in this function)
  743 |     SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
      |                  ^~~~~~~~~~~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c: In function 'POWER_SetPmicMode':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:226:5: warning: implicit declaration of function '__disable_irq'; did you mean 'DisableIRQ'? [-Wimplicit-function-declaration]
  226 |     __disable_irq();
      |     ^~~~~~~~~~~~~
      |     DisableIRQ
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:231:5: warning: implicit declaration of function '__enable_irq' [-Wimplicit-function-declaration]
  231 |     __enable_irq();
      |     ^~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c: In function 'POWER_EnterSleep':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:385:12: warning: implicit declaration of function '__get_PRIMASK' [-Wimplicit-function-declaration]
  385 |     pmsk = __get_PRIMASK();
      |            ^~~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:387:5: error: 'SCB' undeclared (first use in this function)
  387 |     SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
      |     ^~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:387:18: error: 'SCB_SCR_SLEEPDEEP_Msk' undeclared (first use in this function)
  387 |     SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
      |                  ^~~~~~~~~~~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:388:5: warning: implicit declaration of function '__WFI' [-Wimplicit-function-declaration]
  388 |     __WFI();
      |     ^~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:389:5: warning: implicit declaration of function '__set_PRIMASK' [-Wimplicit-function-declaration]
  389 |     __set_PRIMASK(pmsk);
      |     ^~~~~~~~~~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c: In function 'POWER_EnableAutoWake':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:396:24: error: 'PMC_IRQn' undeclared (first use in this function); did you mean 'PMC0_IRQn'?
  396 |     EnableDeepSleepIRQ(PMC_IRQn);
      |                        ^~~~~~~~
      |                        PMC0_IRQn
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c: In function 'initXSPI':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:822:5: warning: implicit declaration of function '__DSB' [-Wimplicit-function-declaration]
  822 |     __DSB();
      |     ^~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:823:5: warning: implicit declaration of function '__ISB' [-Wimplicit-function-declaration]
  823 |     __ISB();
      |     ^~~~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c: In function 'POWER_EnterLowPower_FullConfig':
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:988:5: error: 'SCB' undeclared (first use in this function)
  988 |     SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
      |     ^~~
/opt/samba/nxa06898/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT798S/drivers/fsl_power.c:988:17: error: 'SCB_SCR_SLEEPDEEP_Msk' undeclared (first use in this function)
  988 |     SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
      |                 ^~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

If I want to overpass this I need to put a lot of ifdef in the `fsl_power.h` and `fsl_power.c`
Can we have a power driver just for DSP?

How do you see these errors fixed?

Reply:
Thank you for pointing the solutions. Sorry I ignored HIFI core support before, I have updated the code.